terraform/aws/public-cloud/main.tf

```diff
@@ -154,13 +168,13 @@ module "iam" {
 }
 
 # Generate an etcd URL for the cluster
-resource "template_file" "etcd_discovery_url" {
-  template = "${file("/dev/null")}"
-  provisioner "local-exec" {
-    command = "curl https://discovery.etcd.io/new?size=${var.masters} > ${var.etcd_discovery_url_file}"
-  }
-  # This will regenerate the discovery URL if the cluster size changes
-  vars {
-    size = "${var.masters}"
-  }
+resource "null_resource" "etcd_discovery_url" {
+    provisioner "local-exec" {
+        command = "curl -s https://discovery.etcd.io/new?size=${var.masters} > ${var.etcd_discovery_url_file}"
+    }
+
+    # This will regenerate the discovery URL if the cluster size changes
```

Comment:
There is no latent issue. My comment here refers to the fact that you are retriggering the null resource when the size changes to generate a new token (emulating the current behaviour with "template_file") but not sure this is useful. As per the link above, that's not what you need when you modify the size of a running cluster. So I think the "triggers" here could be omitted as it does not really solve anything. It does not hurt either though